File: app/src/main/java/cn/ifafu/ifafu/data/local/converter/StringMapConverter.java
package cn.ifafu.ifafu.data.local.converter;

import com.alibaba.fastjson.JSONObject;

import org.greenrobot.greendao.converter.PropertyConverter;

import java.util.Map;

public class StringMapConverter implements PropertyConverter<Map<String, Object>, String> {
    @Override
    public Map<String, Object> convertToEntityProperty(String databaseValue) {
        return JSONObject.parseObject(databaseValue).getInnerMap();
    }

    @Override
    public String convertToDatabaseValue(Map<String, Object> entityProperty) {
        return JSONObject.toJSONString(entityProperty);
    }
}

File: app/src/main/java/cn/ifafu/ifafu/data/local/converter/StringMapConverter.kt
package cn.ifafu.ifafu.data.local.converter

import androidx.room.TypeConverter
import com.alibaba.fastjson.JSONObject

class StringMapConverter {
    @TypeConverter
    fun convertToEntityProperty(databaseValue: String): HashMap<String, String> {
        return HashMap(JSONObject.parseObject(databaseValue).innerMap as Map<String, String>)
    }

    @TypeConverter
    fun convertToDatabaseValue(entityProperty: HashMap<String, String>): String {
        return JSONObject.toJSONString(entityProperty)
    }
}